cub/test/catch2_test_memcpy_batched.cu

```diff
@@ -0,0 +1,201 @@
+
+DECLARE_LAUNCH_WRAPPER(cub::DeviceMemcpy::Batched, memcpy_batched);
+
+/**
+ * @brief Function object that takes an offset and returns an iterator at the given
+ * offset relative to a fixed base iterator.
+ */
+template <typename IteratorT>
+struct offset_to_ptr_op
+{
+  IteratorT base_it;
+
+  template <typename T>
+  __host__ __device__ __forceinline__ IteratorT operator()(T offset) const
+  {
+    return base_it + offset;
+  }
+};
+
+/**
+ * @brief Used for generating a shuffled but cohesive sequence of output-buffer offsets for the
+ * sequence of input-buffers.
+ */
+template <typename BufferOffsetT, typename BufferSizeT, typename ByteOffsetT>
+void get_shuffled_buffer_offsets(const c2h::device_vector<BufferSizeT>& buffer_sizes,
+                                 c2h::device_vector<ByteOffsetT>& buffer_offsets,
+                                 c2h::seed_t seed)
+{
+  auto const num_buffers = static_cast<BufferOffsetT>(buffer_sizes.size());
+
+  // We're remapping the i-th buffer to pmt_idxs[i]
+  c2h::device_vector<BufferOffsetT> pmt_idxs(num_buffers);
+  const auto buffer_index_it = thrust::make_counting_iterator(BufferOffsetT{0});
+  thrust::shuffle_copy(
+    buffer_index_it,
+    buffer_index_it + num_buffers,
+    pmt_idxs.begin(),
+    thrust::default_random_engine(static_cast<std::uint32_t>(seed.get())));
+
+  c2h::device_vector<ByteOffsetT> permuted_offsets(num_buffers);
+  auto permuted_buffer_sizes_it = thrust::make_permutation_iterator(buffer_sizes.begin(), pmt_idxs.begin());
+  thrust::exclusive_scan(permuted_buffer_sizes_it, permuted_buffer_sizes_it + num_buffers, permuted_offsets.begin());
+
+  c2h::device_vector<BufferOffsetT> scatter_idxs(num_buffers);
+  thrust::scatter(buffer_index_it, buffer_index_it + num_buffers, pmt_idxs.cbegin(), scatter_idxs.begin());
+
+  // Gather the permuted offsets for shuffled buffer offsets
+  thrust::gather(scatter_idxs.cbegin(), scatter_idxs.cend(), permuted_offsets.cbegin(), buffer_offsets.begin());
+}
+
+C2H_TEST("DeviceMemcpy::Batched works", "[memcpy]")
+try
+{
+  using src_ptr_t = const uint8_t*;
+  using dst_ptr_t = uint8_t*;
+
+  // Type used for indexing into the array of buffers
+  using buffer_offset_t = uint32_t;
+
+  // Type used for indexing into individual bytes of a buffer (large enough to cover the max buffer
+  using buffer_size_t = uint32_t;
+
+  // Type used for indexing into bytes over *all* the buffers' sizes
+  using byte_offset_t = uint32_t;
+
+  // Total number of bytes that are targeted to be copied on each run
+  constexpr buffer_offset_t target_copy_size = 32U << 20;
+
+  // Pairs of [min, max] buffer sizes
+  auto buffer_size_range = GENERATE_COPY(
+    table<int, int>(
+      {std::make_tuple(0, 1),
+       std::make_tuple(1, 2),
+       std::make_tuple(0, 32),
+       std::make_tuple(1, 1024),
+       std::make_tuple(1, 32 * 1024),
+       std::make_tuple(128 * 1024, 256 * 1024),
+       std::make_tuple(target_copy_size, target_copy_size)}),
```

Comment:
Can we just use initializer lists instead of `make_tuple`?
```suggestion
      {{0, 1},
       {1, 2},
       {0, 32},
       {1, 1024},
       {1, 32 * 1024},
       {128 * 1024, 256 * 1024},
       {target_copy_size, target_copy_size}}),
```

Reply:
Thought it hadn't worked when I first tried. Apparently I was wrong. Switched to using init lists now.